src/dataspec/base.py

```diff
@@ -125,6 +125,38 @@ def with_details(self, tag: Tag, loc: Any = NO_ERROR_PATH) -> "ErrorDetails":
             self.path.insert(0, loc)
         return self
 
+    def as_map(self) -> Mapping[str, Union[str, List[str]]]:
```

Comment:
You may be able to use `TypedDict` here if you want.

Reply:
That would be nice, but right now we support Python 3.6+ and `TypedDict` is 3.8+.

MyPy raises a bunch of errors if you try to get around it like this:

```python
try:
    from typing import TypedDict
except ImportError:
    ErrorDetailsMap = Mapping[str, Union[str, List[str]]]
else:
    class ErrorDetailsMap(TypedDict):
        ...
```